gui/wxpython/gui_core/goutput.py

```diff
@@ -473,54 +465,27 @@ def OnCmdProgress(self, event):
         self.progressbar.SetValue(event.value)
         event.Skip()
 
-    def CmdProtocolSave(self):
-        """Save list of manually entered commands into a text log file"""
-        if self.cmdFileProtocol is None:
-            return  # it should not happen
-
-        try:
-            with open(self.cmdFileProtocol, "a") as output:
-                cmds = self.cmdPrompt.GetCommands()
-                output.write("\n".join(cmds))
-                if len(cmds) > 0:
-                    output.write("\n")
-        except IOError as e:
-            GError(
-                _("Unable to write file '{filePath}'.\n\nDetails: {error}").format(
-                    filePath=self.cmdFileProtocol, error=e
-                )
-            )
-
-        self.showNotification.emit(
-            message=_("Command log saved to '{}'".format(self.cmdFileProtocol))
+    def OnCmdExportHistory(self, event):
+        """Export the history of executed commands stored
+        in a .wxgui_history file to a selected file."""
+        dlg = wx.FileDialog(
+            self,
+            message=_("Save file as..."),
+            defaultFile="grass_cmd_log.txt",
+            wildcard=_("{txt} (*.txt)|*.txt|{files} (*)|*").format(
+                txt=_("Text files"), files=_("Files")
+            ),
+            style=wx.FD_SAVE | wx.FD_OVERWRITE_PROMPT,
         )
-        self.cmdFileProtocol = None
-
-    def OnCmdProtocol(self, event=None):
-        """Save commands into file"""
-        if not event.IsChecked():
-            # stop capturing commands, save list of commands to the
-            # protocol file
-            self.CmdProtocolSave()
-        else:
-            # start capturing commands
-            self.cmdPrompt.ClearCommands()
-            # ask for the file
-            dlg = wx.FileDialog(
-                self,
-                message=_("Save file as..."),
-                defaultFile="grass_cmd_log.txt",
-                wildcard=_("%(txt)s (*.txt)|*.txt|%(files)s (*)|*")
-                % {"txt": _("Text files"), "files": _("Files")},
-                style=wx.FD_SAVE,
-            )
-            if dlg.ShowModal() == wx.ID_OK:
-                self.cmdFileProtocol = dlg.GetPath()
-            else:
-                wx.CallAfter(self.btnCmdProtocol.SetValue, False)
 
-            dlg.Destroy()
+        if dlg.ShowModal() == wx.ID_OK:
+            path = dlg.GetPath()
+            self.cmdPrompt.CopyHistory(path)
+            self.showNotification.emit(
+                message=_("Command history saved to '{}'".format(path))
+            )
```

Comment:
Here you emit the notification even when there was an error, so you probably need to return true/false from CopyHistory method.